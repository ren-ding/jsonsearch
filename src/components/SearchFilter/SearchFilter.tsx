import React, { Component } from 'react';
import Dropdown, {Option} from 'react-dropdown';
import 'react-dropdown/style.css';
import './style/search-filter.css';

export interface SearchFilterPropsInterface {
    searchTableDropdownList: Array<Option>,
    onChange: (option:Option) => void,
    selectedSearchTable: number,
    onSearchableFieldsChange: (event: {target: { value: string }}) => void,
    onSearchValueChange: (event: {target: { value: string }}) => void
}

interface SearchFilterStatesInterface {

}

interface OnInputChangeInterface {
    labelName: string,
    onInputChange:(event: {target: { value: string }}) => void
}
  
const InputBox = ({
    labelName,
    onInputChange
}:OnInputChangeInterface)=>(
    <div className='ui fluid icon input'>
        <input type="text" size={30} onChange={onInputChange} placeholder={labelName}></input>
        <i className='search icon'></i>
    </div>
);

export default class SearchFilter extends Component<SearchFilterPropsInterface, SearchFilterStatesInterface> {
    constructor(props:SearchFilterPropsInterface) {
        super(props);
    }

    render() {
        return (
        <div className='search-filter-wrapper'>
            <div className='drop-down-wrapper'>
                <Dropdown 
                    options={this.props.searchTableDropdownList}
                    onChange={this.props.onChange}
                    value={this.props.selectedSearchTable.toString()} 
                />
            </div>
            <div className='searchable-fields-input-box-wrapper'>
                <InputBox
                    labelName = "Search fields (e.g. name,tags): "
                    onInputChange={this.props.onSearchableFieldsChange}
                />
            </div>
            <div className='search-value-input-box-wrapper'>
                <InputBox
                    labelName = "Search value"
                    onInputChange={this.props.onSearchValueChange}
                />
            </div>
        </div>);
    }
}