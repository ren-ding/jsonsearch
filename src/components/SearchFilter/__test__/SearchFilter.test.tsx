import React from 'react';
import SearchFilter, {SearchFilterPropsInterface} from '../SearchFilter';
import renderer from 'react-test-renderer';

describe('SearchFilter', () => {
    let props:SearchFilterPropsInterface;
    beforeEach(()=> {
        props = {
            searchTableDropdownList: [],
            onChange: () => {},
            selectedSearchTable: 0,
            onSearchableFieldsChange: (event: {target: { value: string }}) => {},
            onSearchValueChange: (event: {target: { value: string }}) => {}
        }
    });

    it('should render to match snapshot', () => {
        const tree = renderer.create(<SearchFilter {...props}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});