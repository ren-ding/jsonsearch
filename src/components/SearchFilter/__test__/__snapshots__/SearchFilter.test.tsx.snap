// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter should render to match snapshot 1`] = `
<div
  className="search-filter-wrapper"
>
  <div
    className="drop-down-wrapper"
  >
    <div
      className="Dropdown-root"
    >
      <div
        aria-haspopup="listbox"
        className="Dropdown-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="Dropdown-placeholder is-selected"
        >
          0
        </div>
        <div
          className="Dropdown-arrow-wrapper"
        >
          <span
            className="Dropdown-arrow"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="searchable-fields-input-box-wrapper"
  >
    <div
      className="ui fluid icon input"
    >
      <input
        onChange={[Function]}
        placeholder="Search fields (e.g. name,tags): "
        size={30}
        type="text"
      />
      <i
        className="search icon"
      />
    </div>
  </div>
  <div
    className="search-value-input-box-wrapper"
  >
    <div
      className="ui fluid icon input"
    >
      <input
        onChange={[Function]}
        placeholder="Search value"
        size={30}
        type="text"
      />
      <i
        className="search icon"
      />
    </div>
  </div>
</div>
`;
